db_revisions/versions/ccc50ec18a7e_add_submission_upload_malformed_to_state_enum.py

```diff
@@ -0,0 +1,51 @@
+"""add SUBMISSION_UPLOAD_MALFORMED to state enum
+
+Revision ID: ccc50ec18a7e
+Revises: 11dfc8200daa
+Create Date: 2024-04-07 01:02:13.310717
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = "ccc50ec18a7e"
+down_revision: Union[str, None] = "ffd779216f6d"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+old_options = (
+    "SUBMISSION_UPLOADED",
+    "VALIDATION_IN_PROGRESS",
+    "VALIDATION_WITH_ERRORS",
+    "VALIDATION_WITH_WARNINGS",
+    "VALIDATION_SUCCESSFUL",
+    "SUBMISSION_ACCEPTED",
+    "SUBMISSION_STARTED",
+)
+new_options = sorted(old_options + ("SUBMISSION_UPLOAD_MALFORMED",))
+
+
+def upgrade() -> None:
+    with op.batch_alter_table("submission", schema=None) as batch_op:
+        batch_op.alter_column(
+            "state",
+            type_=sa.Enum(*new_options, name="submissionstate"),
+            existing_type=sa.Enum(*old_options, name="submissionstate"),
+            existing_server_default=sa.text("'text'"),
```

Comment:
just curious is this is needed, and if it is why it's set to the literal 'text', or is that just a placeholder? looks like `existing_server_default`, and `existing_nullable` are MySQL specific, or does it serve a purpose for us?

Reply:
Yeah unfortunately because sqlite and postgres you have to do the following to update the submissionstate enum:
```
def upgrade() -> None:
    if "sqlite" not in context.get_context().dialect.name:
        op.execute("ALTER TYPE submissionstate RENAME TO submissionstate_old")
        op.execute(f"CREATE TYPE submissionstate AS ENUM{new_options}")
        op.execute("ALTER TABLE submission ALTER COLUMN state TYPE submissionstate USING state::text::submissionstate")
        op.execute("DROP TYPE submissionstate_old")
    op.add_column("submission", sa.Column("accepter", sa.String))


def downgrade() -> None:
    if "sqlite" not in context.get_context().dialect.name:
        op.execute("ALTER TYPE submissionstate RENAME TO submissionstate_old")
        op.execute(f"CREATE TYPE submissionstate AS ENUM{old_options}")
        op.execute("ALTER TABLE submission ALTER COLUMN state TYPE submissionstate USING state::text::submissionstate")
        op.execute("DROP TYPE submissionstate_old")
    op.drop_column("submission", "accepter")
```
Otherwise it might work fine for pytests, but in actual functional runs the new value won't exist.

Reply:
Will modify it.